From: Ashok Ramakrishnan <ashok.ramakrishnan@simplivity.com>
Date: Tue, 16 Aug 2016 13:34:45 +0000
Subject: HYP-none : Fix compile failure when using configure.developer.

Having a variable declared inside a for loop is prohibited in debug build.
---
 source3/locking/leases_db.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/source3/locking/leases_db.c b/source3/locking/leases_db.c
index 04020ee..6d40874 100644
--- a/source3/locking/leases_db.c
+++ b/source3/locking/leases_db.c
@@ -32,6 +32,7 @@
 
 /* the leases database handle */
 static struct db_context *leases_db;
+extern bool smbXsrv_lookup_persistent_id(uint64_t);
 
 void remove_stale_lease_entries(struct leases_db_value *d)
 {
@@ -59,6 +60,7 @@ static int leases_db_traverse_persist_fn(struct db_record *rec, void *_state)
 	struct leases_db_value *d;
 	bool found_persistent_open = False;
 	NTSTATUS status;
+	struct leases_db_file *entry;
 
 	key = dbwrap_record_get_key(rec);
 	value = dbwrap_record_get_value(rec);
@@ -91,7 +93,7 @@ static int leases_db_traverse_persist_fn(struct db_record *rec, void *_state)
 
 	for (i=0; i<d->num_files; i++) {
 		DEBUG(1, ("leases_db_traverse_persist_fn: Loop iteration %i\n", i));
-		struct leases_db_file *entry = &d->files[i];
+		entry = &d->files[i];
 		if ( entry->open_persistent_id != UINT64_MAX && smbXsrv_lookup_persistent_id(entry->open_persistent_id) ) {
 			entry->stale = false; /* [skip] in idl */
 			found_persistent_open = True;
